test/azul_test_case.py

```diff
@@ -82,8 +90,12 @@ def dummy_get_credentials(self):
         botocore.session.Session.get_credentials = dummy_get_credentials
         boto3.session.Session.get_credentials = dummy_get_credentials
 
+        # Ensure that mock leakages fail by targeting another region
```

Comment:
I'd prefer the patch object not to be global. You should follow existing precedent (as usual), init to None at class scope and construct the patch instance here. This eliminates the need to think about reusing the `patch` instance and lets you consolidate the comments into one place in the code base.